app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached = {}
+
+    def inner(*args, **kwargs) -> Any:
+        if args in cached:
+            print("Getting from cache")
+        else:
+            new_cache = func(*args)
+            cached[args] = new_cache
+            print("Calculating new result")
+
+        return func(*args) if args not in cached.keys() else cached[args]
```

Comment:
and may be better use just one if else statment

Reply:
If  I replace one if statement, func is going to have two returns, isn't it?

Reply:
I've found out how to deal with that. Now It looks better.